spec/acceptance/command_shell_spec.rb

```diff
@@ -0,0 +1,533 @@
+          # @return [String] The file contents if found
+          def get_file_attachment_contents(path)
+            return 'none resent' unless File.exist?(path)
+
+            content = File.binread(path)
+            content.blank? ? 'file created - but empty' : content
+          end
+
+          before :each do |example|
+            next unless example.respond_to?(:parameter)
+
+            # Add the test environment metadata to the rspec example instance - so it appears in the final allure report UI
+            test_environment.each do |key, value|
+              example.parameter(key, value)
+            end
+          end
+
+          after :all do
+            driver.close_payloads
+            console.reset
+          end
+
+          context "#{Acceptance::Session.current_platform}" do
+            describe "#{Acceptance::Session.current_platform}/#{command_shell_runtime_name} command shell successfully opens a session for the #{payload_config[:name].inspect} payload" do
+              it(
+                "exposes available metasploit commands",
+                if: (
+                  # Assume that regardless of payload, staged/unstaged/etc, the command shell will have the same commands available
+                  # So only run this test when config_index == 0
+                  payload_config_index == 0 && Acceptance::Session.supported_platform?(payload_config)
+                  # Run if ENV['METERPRETER'] = 'java php' etc
+                  Acceptance::Session.run_session?(command_shell_config) &&
+                    # Only run payloads / tests, if the host machine can run them
+                    Acceptance::Session.supported_platform?(payload_config)
+                )
+              ) do
+                begin
+                  replication_commands = []
+                  current_payload_status = ''
+
+                  # Ensure we have a valid session id; We intentionally omit this from a `before(:each)` to ensure the allure attachments are generated if the session dies
+                  payload_process, session_id = payload_process_and_session_id
+                  expect(payload_process).to(be_alive, proc do
+                    current_payload_status = "Expected Payload process to be running. Instead got: payload process exited with #{payload_process.wait_thread.value} - when running the command #{payload_process.cmd.inspect}"
+
+                    Allure.add_attachment(
+                      name: 'Failed payload blob',
+                      source: Base64.strict_encode64(File.binread(payload_process.payload_path)),
+                      type: Allure::ContentType::TXT
+                    )
+
+                    current_payload_status
+                  end)
+                  expect(session_id).to_not(be_nil, proc do
+                    "There should be a session present"
+                  end)
+
+                  resource_command = "resource scripts/resource/meterpreter_compatibility.rc"
```

Comment:
Looking at this I'm wondering should this resource script also be renamed 🤔 

It has four matches in across framework(one being a comment):
![image](https://github.com/user-attachments/assets/cfc3660e-872f-4ecf-9017-73f3b0a02e51)

Reply:
https://github.com/rapid7/metasploit-framework/pull/19413/files#r1768565200